generation/new_client_hermetic_build/new-client.py

```diff
@@ -0,0 +1,253 @@
+    "then you specify this value as 'google/maps/routing'",
+)
+@click.option(
+    "--product-docs",
+    required=True,
+    type=str,
+    prompt="Product Documentation URL",
+    help="Documentation URL that appears in README.md",
+)
+@click.option(
+    "--rest-docs",
+    type=str,
+    help="If it exists, link to the REST Documentation for a service",
+)
+@click.option(
+    "--rpc-docs",
+    type=str,
+    help="If it exists, link to the RPC Documentation for a service",
+)
+@click.option(
+    "--api-description",
+    required=True,
+    type=str,
+    prompt="Description for README. The first sentence is prefixed by the "
+    "pretty name",
+    help="Description that appears in README.md",
+)
+@click.option(
+    "--transport",
+    type=click.Choice(["grpc", "http", "both"]),
+    default="grpc",
+    show_default=True,
+    help="A label that appears in repo-metadata.json",
+)
+@click.option(
+    "--destination-name",
+    type=str,
+    default=None,
+    help="The directory name of the new library. By default it's "
+    "java-<api_shortname>",
+)
+@click.option(
+    "--distribution-name",
+    type=str,
+    help="Maven coordinates of the generated library. By default it's "
+    "com.google.cloud:google-cloud-<api_shortname>",
+)
+@click.option(
+    "--release-level",
+    type=click.Choice(["stable", "preview"]),
+    default="preview",
+    show_default=True,
+    help="A label that appears in repo-metadata.json. The first library "
+         "generation is always 'preview'."
+)
+@click.option(
+    "--api-id",
+    type=str,
+    help="The value of the apiid parameter used in README.md It has link to "
+         "https://console.cloud.google.com/flows/enableapi?apiid=<api_id>"
+)
+@click.option(
+    "--requires-billing",
+    type=bool,
+    default=True,
+    show_default=True,
+    help="Based on this value, README.md explains whether billing setup is "
+         "needed or not."
+)
+@click.option(
+    "--cloud-api",
+    type=bool,
+    default=True,
+    show_default=True,
+    help="If true, the artifact ID of the library is 'google-cloud-'; "
+         "otherwise 'google-'"
+)
+@click.option(
+    "--group-id",
+    type=str,
+    default="com.google.cloud",
+    show_default=True,
+    help="The group ID of the artifact when distribution name is not set"
+)
+@click.option(
+    "--library-type",
+    type=str,
+    default="GAPIC_AUTO",
+    show_default=True,
+    help="A label that appears in repo-metadata.json to tell how the library is "
+         "maintained or generated"
+)
+def add_new_library(
+    api_shortname,
+    name_pretty,
+    proto_path,
+    product_docs,
+    rest_docs,
+    rpc_docs,
+    api_description,
+    transport,
+    destination_name,
+    distribution_name,
+    release_level,
+    api_id,
+    requires_billing,
+    cloud_api,
+    group_id,
+    library_type,
+):
+    cloud_prefix = "cloud-" if cloud_api else ""
```

Comment:
Why do we need `if` statements since we should get all parameters from user?

I expect some simple assignments.

Reply:
I agree we can get rid of these statements. It came from an initial verbatim copy of the original new-client. I'll take a second look at these statements